### PR TITLE
Continuous Integration

### PR DESCRIPTION
**Pull request by Nemo157**
I assume someone with permission would have to enable this repository in Travis CI, not sure what exact permission that would be though.

### CODE REVIEW

**Comment by wking** [comment]
> I assume someone with permission would have to enable this repository in Travis CI, not sure what exact permission that would be though.

Looks like it's an org admin ([docs][1]).  Travis has some docs [here][2], but I'm not sure if that's for the new [GitHub App][3] approach.  Perhaps an org admin needs to go through the process starting [here][4] for the $0 open source plan?  @carols10cents, could you look into setting this up at the org level? 

[1]: https://developer.github.com/apps/differences-between-apps/#who-can-install-github-apps-and-authorize-oauth-apps
[2]: https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI
[3]: https://developer.github.com/changes/2018-04-25-github-services-deprecation/ 
[4]: https://github.com/marketplace/travis-ci/

File: .travis.yml
Comment (wking) [inline .travis.yml:6]: I think we want to include `- os: linux` as well.

**Comment by wking** [comment]
Moving the fetcher to a separate, unpublished crate makes sense to me, and seems orthogonal to CI.  Can you spin it out into a separate PR?

**Comment by Nemo157** [comment]
Sure

File: .travis.yml
Comment (Nemo157) [reply .travis.yml:6]: The default is linux, this adds an extra build that is run against macOS on top of those. You can see the full build matrix here: https://travis-ci.org/Nemo157/license-exprs/builds/376051360

Comment (Nemo157) [reply .travis.yml:6]: How it actually works, each entry in the `include` array defines one job, the first job here overrides the `os` to be `osx` and uses the default `rust` version, the other 3 jobs each override the `rust` version and use the default `os` of `linux`.

**Review by wking — APPROVED** [review]
This looks good to me.  Unless someone weighs in with additional issues, I'll merge it once @carols10cents has time to [enable Travis for the org][1].

[1]: https://github.com/rust-lang-nursery/license-exprs/pull/22#issuecomment-387174092

**Comment by carols10cents** [comment]
I think I did it? https://travis-ci.com/rust-lang-nursery/license-exprs

**Comment by wking** [comment]
> I think I did it? https://travis-ci.com/rust-lang-nursery/license-exprs

Yup, looks good to me.  I've merged this branch, and the new master is running through its tests [here][1].  Maybe I should have kicked Travis somehow before merging to make sure everything passed, but it's too late now ;).

[1]: https://travis-ci.com/rust-lang-nursery/license-exprs/builds/72738250

**Comment by Nemo157** [comment]
I guess it's under travis-ci.com instead of travis-ci.org because it's using the new check application support? I wonder if all the other `rust-lang-nursery` projects should migrate: https://travis-ci.org/rust-lang-nursery

**Comment by wking** [comment]
> I guess it's under travis-ci.com instead of travis-ci.org because it's using the new check application support?

Yeah, see [here][1] and [here][2].

> I wonder if all the other rust-lang-nursery projects should migrate: https://travis-ci.org/rust-lang-nursery

They have [a migration tool][3], if folks want to do that in bulk.  And without that, you can also [opt-in to having Travis migrate you][4].

[1]: https://blog.travis-ci.com/2018-05-02-open-source-projects-on-travis-ci-com-with-github-apps
[2]: https://docs.travis-ci.com/user/open-source-on-travis-ci-com/
[3]: https://github.com/travis-ci/travis_migrate_to_apps
[4]: https://docs.travis-ci.com/user/open-source-on-travis-ci-com/#Existing-Open-Source-Repositories-on-travis-ci.org

**Pull request merged**
